docker/zisk/Dockerfile

```diff
@@ -37,14 +38,21 @@ RUN wget https://developer.download.nvidia.com/compute/cuda/repos/$(. /etc/os-re
     apt install -y cuda-toolkit && \
     apt-get clean && rm -rf /var/lib/apt/lists/*
 
+# If current environment is in CI or not.
+ARG CI
+
 # Copy the ZisK SDK installer script from the workspace context
 COPY scripts/sdk_installers/install_zisk_sdk.sh /tmp/install_zisk_sdk.sh
 RUN chmod +x /tmp/install_zisk_sdk.sh
 
 # Run the ZisK SDK installation script using ziskup.
-# This script installs the 'zisk' Rust toolchain and cargo-zisk.
-# TODO: Download the proving key if the CI runner has enough disk space.
-RUN SETUP_KEY=verify /tmp/install_zisk_sdk.sh && rm /tmp/install_zisk_sdk.sh # Clean up the script
+# This script installs the 'zisk' Rust toolchain and `cargo-zisk`
+#
+# If argument `CI` is set, we only install verifying key, this is used by github
+# CI runner which only has small disk space (proving key requires ~35 GB).
+RUN if [ -n "$CI" ]; then export SETUP_KEY=verify; fi && \
+    /tmp/install_zisk_sdk.sh && \
+    rm /tmp/install_zisk_sdk.sh # Clean up the script
```

Comment:
Asking to understand more:
1. These verifying and proving keys are related to aggregation circuits?
2. What's the consequence of only installing one key but not the other?
3. Maybe with the devops GH runners we can remove this since they have 100GiB?

Sorry some are basic questions :)

Reply:
> Reg 1. I think I was mostly thinking on proving and verifying keys that usually depend on the program you want to prove.

It indeed depends on their zkVM circuit and the aggregation circuit I think, it does some pre-processing when installation that makes the proving faster but takes huge disk space. Tho it doesn't depend on the guest program because guest program could be seen as a private input to the zkVM circuit.

Reply:
Ah sorry, I think I was sticking to other non-zkVM thinking where guest programs were custom -- in this case the VM is the program and the "guest programs" are inputs yes.